cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -773,8 +774,19 @@ public boolean accept(String className, URL classUrl, URL classPathUrl) {
       LOG.warn("Not including HBaseTableUtil classes in submitted Job Jar since they are not available");
     }
 
+    // Add the logback.xml as a resource while creating the MapReduce Job JAR
+    Set<URI> logbackURI = Sets.newHashSet();
+    try {
+      URL logback = getClass().getResource("logback.xml");
```

Comment:
Have you verify? `Class.getResource()` will prepend the current class package name as the prefix. Hence it will try to look for a resource in "co/cask/cdap/internal/app/runtime/batch/logback.xml", which I don't think there is such file in the classloader path.